x-pack/elastic-agent/pkg/reporter/reporter.go

```diff
@@ -64,46 +72,10 @@ func (r *Reporter) Close() {
 	}
 }
 
-// OnStarting reports application starting event.
-func (r *Reporter) OnStarting(ctx context.Context, application string) {
-	msg := fmt.Sprintf("Application: %s[%s]: State change: STARTING", application, r.info.AgentID())
-	rec := generateRecord(EventTypeState, EventSubTypeStarting, msg)
-	r.report(ctx, rec)
-}
-
-// OnRunning reports application running event.
-func (r *Reporter) OnRunning(ctx context.Context, application string) {
-	msg := fmt.Sprintf("Application: %s[%s]: State change: IN_PROGRESS", application, r.info.AgentID())
-	rec := generateRecord(EventTypeState, EventSubTypeInProgress, msg)
-	r.report(ctx, rec)
-}
-
-// OnFailing reports application failed event.
-func (r *Reporter) OnFailing(ctx context.Context, application string, err error) {
-	msg := fmt.Sprintf("Application: %s[%s]: %v", application, r.info.AgentID(), err)
-	rec := generateRecord(EventTypeError, EventSubTypeConfig, msg)
-	r.report(ctx, rec)
-}
-
-// OnStopping reports application stopped event.
-func (r *Reporter) OnStopping(ctx context.Context, application string) {
-	msg := fmt.Sprintf("Application: %s[%s]: State change: STOPPING", application, r.info.AgentID())
-	rec := generateRecord(EventTypeState, EventSubTypeStopping, msg)
-	r.report(ctx, rec)
-}
-
-// OnStopped reports application stopped event.
-func (r *Reporter) OnStopped(ctx context.Context, application string) {
-	msg := fmt.Sprintf("Application: %s[%s]: State change: STOPPED", application, r.info.AgentID())
-	rec := generateRecord(EventTypeState, EventSubTypeStopped, msg)
-	r.report(ctx, rec)
-}
-
-// OnFatal reports applications fatal event.
-func (r *Reporter) OnFatal(ctx context.Context, application string, err error) {
-	msg := fmt.Sprintf("Application: %s[%s]: %v", application, r.info.AgentID(), err)
-	rec := generateRecord(EventTypeError, EventSubTypeConfig, msg)
-	r.report(ctx, rec)
+// OnStateChange called when state of an application changes.
+func (r *Reporter) OnStateChange(id string, name string, state state.State) {
```

Comment:
i was passing context down the reporter so we can eventually control timeout of reporting or call it within time sensitive operations correctly without blocking.
passing context as a property serves as a cancellation token for long running loops, which is not present here

Reply:
Reporter has always been created with a context, but that context was never used. So I used that context, instead of passing the context from the application into the reporter. That way all state changes go to the reporter and are not cancelled unless the entire reporter is cancelled.